test_runner/regress/test_lsn_mapping.py

```diff
@@ -108,6 +111,64 @@ def test_lsn_mapping(neon_env_builder: NeonEnvBuilder):
         assert Lsn(result["lsn"]) >= last_flush_lsn
 
 
+def test_get_lsn_by_timestamp_cancelled(neon_env_builder: NeonEnvBuilder):
+    """
+    Test if cancelled pageserver get_lsn_by_timestamp request is correctly handled.
+    Added as an effort to improve error handling and avoid full anyhow backtrace.
+    """
+
+    env = neon_env_builder.init_start()
+    env.pageserver.allowed_errors.extend(
+        [
+            ".*request was dropped before completing.*",
+            ".*Cancelled request finished with an error: Cancelled\n",
+        ]
+    )
+
+    client = env.pageserver.http_client()
+    failpoint = "find-lsn-for-timestamp-pausable"
+    client.configure_failpoints((failpoint, "pause"))
+
+    with env.endpoints.create_start("main") as endpoint:
+        conn = endpoint.connect()
+        cur = conn.cursor()
+        cur.execute("CREATE TABLE foo (x integer)")
+        cur.execute("INSERT INTO foo VALUES(0)")
+        # Get the timestamp at UTC
+        probe_timestamp = query_scalar(cur, "SELECT clock_timestamp()").replace(tzinfo=None)
+        cur.execute("INSERT INTO foo VALUES (-1)")
```

Comment:
The `conn` and `cur` are also context managers. We have a default mode of autocommit=True but I recall being confused by the different ways in different kind of uses.

However, there's a bigger problem, assuming we want to continue keeping this `endpoint` configuration:

When these statements complete, from the point of view of the system under test (compute, safekeeper, pageserver) it is indeterminate how much of the WAL (your actions with the `endpoint`) has arrived at pageserver.

Usually we use `wait_for_last_(flush|record)_lsn` to make sure the wal arrives to pageserver.

But as said, in this case we do not need any changes in the timeline database, so we could just query with timestamp 0.